schema_enforcer/schemas/validator.py

```diff
@@ -84,27 +86,110 @@ def validate(self, data: dict, strict: bool):  # pylint: disable=W0613
             self.add_validation_error(self.error)
 
 
+class PydanticValidation(BaseValidation):
+    """Basic wrapper for Pydantic models to be used as validators."""
+
+    model: BaseModel
+
+    def validate(self, data: dict, strict: bool = False):
+        """Validate data against Pydantic model.
+
+        Args:
+            data (dict): variables to be validated by validator
+            strict (bool): true when --strict cli option is used to request strict validation (if provided)
+
+        Returns:
+            None
+
+        Use add_validation_error and add_validation_pass to report results.
+        """
+        try:
+            self.model.model_validate(data, strict=strict)
+            self.add_validation_pass()
+        except ValidationError as err:
+            self.add_validation_error(str(err))
+
+
 def is_validator(obj) -> bool:
     """Returns True if the object is a BaseValidation or JmesPathModelValidation subclass."""
     try:
-        return issubclass(obj, BaseValidation) and obj not in (JmesPathModelValidation, BaseValidation)
+        return (issubclass(obj, BaseValidation) or issubclass(obj, BaseModel)) and obj not in (
+            BaseModel,
+            BaseValidation,
+            JmesPathModelValidation,
+        )
     except TypeError:
         return False
 
 
-def load_validators(validator_path: str) -> dict[str, BaseValidation]:
-    """Load all validator plugins from validator_path."""
+def pydantic_validation_factory(orig_model) -> PydanticValidation:
+    """Create a PydanticValidation instance from a Pydantic model."""
+    return type(
+        orig_model.__name__,
+        (PydanticValidation,),
+        {
+            "id": f"{orig_model.id}",
+            "top_level_properties": set([property for property in orig_model.model_fields]),
+            "model": orig_model,
+        },
```

Comment:
Intentional that only these specific properties carry over from the original model? Or should this be more expansive, maybe having specific excluded keys from `orig_model.__dict__` if needed?

Reply:
Yeah, `__dict__` would be much more extensive. I don't know if necessary/worthwhile - on first read I was thinking that this was creating a subclass of `orig_model` but on reread I gather you're doing more of a wrapper class around it, so the current implementation may be just fine?

Reply:
Yes it only wraps the Pydantic model so it can use that model within the `validate` method of `PydanticValidation`.